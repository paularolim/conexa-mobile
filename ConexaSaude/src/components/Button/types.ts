import { TouchableOpacityProps } from 'react-native';

export type ButtonType = 'filled' | 'ghost';

export interface ButtonProps extends TouchableOpacityProps {
  children: string;
  type?: ButtonType;
}
