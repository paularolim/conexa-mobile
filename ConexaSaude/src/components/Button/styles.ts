import { TouchableOpacity } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

import { ButtonType } from './types';

export const Container = styled(TouchableOpacity)<{ type: ButtonType }>`
  background-color: ${({ theme, type }) =>
    type === 'filled' ? theme.colors.green : theme.colors.white};
  border-radius: 4px;
  padding: 8px 16px;
  max-width: 100%;
  justify-content: center;
  align-items: center;
`;

export const Label = styled(Text)<{ type: ButtonType }>`
  color: ${({ theme, type }) => (type === 'filled' ? theme.colors.white : theme.colors.blue)};
  font-size: 18px;
`;
