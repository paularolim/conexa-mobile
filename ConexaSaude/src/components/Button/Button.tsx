import { Container, Label } from './styles';
import { ButtonProps } from './types';

export function Button({ type = 'filled', children, ...rest }: ButtonProps) {
  return (
    <Container type={type} {...rest}>
      <Label type={type}>{children}</Label>
    </Container>
  );
}
