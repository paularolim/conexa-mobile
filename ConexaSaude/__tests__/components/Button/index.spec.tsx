import renderer from 'react-test-renderer';

import { Button } from '@components/Button';

import { Wrapper } from '../../Wrapper';

describe('Button component', () => {
  it('should render default component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Button>Button component</Button>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render type="ghost" component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Button type="ghost">Button component</Button>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
