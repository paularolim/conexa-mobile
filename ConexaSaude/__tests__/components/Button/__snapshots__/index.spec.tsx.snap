// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render default component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2ED47A",
      "borderRadius": 4,
      "justifyContent": "center",
      "maxWidth": "100%",
      "minHeight": 48,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "color": "#FFFFFF",
          "fontSize": 18,
        },
      ]
    }
    type="filled"
  >
    Button component
  </Text>
</View>
`;

exports[`Button component should render type="ghost" component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 4,
      "justifyContent": "center",
      "maxWidth": "100%",
      "minHeight": 48,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "color": "#109CF1",
          "fontSize": 18,
        },
      ]
    }
    type="ghost"
  >
    Button component
  </Text>
</View>
`;
